### PR TITLE
booting-qemu-image: Add static ip configuration

### PR DESCRIPTION
**Pull request by BassemMohsen**

### CODE REVIEW

File: docs/articles/baseplatform/booting-qemu-image.rst [hunk 1]
```diff
@@ -23,9 +23,12 @@ the emulator tool and its dependencies:
 
 Launching the virtual machine
 -----------------------------
+QEMU can be launched with different network backend configurations.
 
-QEMU emulator can be launched with a serial console support using the
-following command:
+Port forwarding configuration:
```
Comment (mardy) [inline docs/articles/baseplatform/booting-qemu-image.rst]: I would remove all the ":" at the end of the section titles

File: docs/articles/baseplatform/booting-qemu-image.rst [hunk 2]
```diff
@@ -23,9 +23,12 @@ the emulator tool and its dependencies:
...
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+QEMU emulator can be launched with a serial console support and port forwarding
```
Comment (mardy) [inline docs/articles/baseplatform/booting-qemu-image.rst]: s/a //

File: docs/articles/baseplatform/booting-qemu-image.rst [hunk 3]
```diff
@@ -41,6 +44,56 @@ following command:
       -m 512 \
       --append "vga=0 uvesafb.mode_option=1280×720-32 root=/dev/hda console=ttyS0 rw mem=512M oprofile.timer=1 " -serial stdio
 
+Static IP address configuration:
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Static IP address facilitates more network services for development and
```
Comment (mardy) [inline docs/articles/baseplatform/booting-qemu-image.rst]: "A static IP address facilitates development of automation and testing scripts."

File: docs/articles/baseplatform/booting-qemu-image.rst [hunk 4]
```diff
@@ -41,6 +44,56 @@ following command:
...
+testing scripts.
+The next commands assumes target ip address is ``192.168.7.2`` and host machine
```
Comment (mardy) [inline docs/articles/baseplatform/booting-qemu-image.rst]: s/is //

File: docs/articles/baseplatform/booting-qemu-image.rst [hunk 5]
```diff
@@ -41,6 +44,56 @@ following command:
...
+   sudo kvm -kernel tmp/deploy/images/qemux86-64/bzImage \
+      -device virtio-net-pci,netdev=net0,mac=52:54:00:12:34:02 -netdev tap,id=net0,ifname=tap0,script=no,downscript=no \
+      -cpu Broadwell \
+      -smp 4 \
+      -hda tmp/deploy/images/qemux86-64/core-image-pelux-minimal-qemux86-64.ext4 \
+      -vga qxl \
+      -no-reboot \
+      -soundhw ac97 \
+      -m 512 \
+      --append "vga=0 uvesafb.mode_option=1280×720-32 root=/dev/hda console=ttyS0 rw mem=512M oprofile.timer=1 ip=192.168.7.2::192.168.7.1:255.255.255.0" -serial stdio
```
Comment (mardy) [inline docs/articles/baseplatform/booting-qemu-image.rst]: s/--append/-append/

File: docs/articles/baseplatform/booting-qemu-image.rst [hunk 6]
```diff
@@ -41,6 +44,56 @@ following command:
...
+
+   route add default gw 192.168.7.1
+
+
+2. Configure the ``tap0`` interface on host machine by executing the following
+commands on the host machine:
+
+.. code-block:: bash
+
+   sudo ip address add 192.168.7.1/24 broadcast 192.168.7.255 dev tap0  \
```
Comment (mardy) [inline docs/articles/baseplatform/booting-qemu-image.rst]: Do you really need the "\\" at the end of the line?

File: docs/articles/baseplatform/booting-qemu-image.rst [hunk 7]
```diff
@@ -41,6 +44,56 @@ following command:
...
+   sudo ip link set dev tap0 up
+   
+
+3. (Optional) To enable the tap device to access internet.
```
Comment (mardy) [inline docs/articles/baseplatform/booting-qemu-image.rst]: "...access internet run..." (without breaking the sentence)

File: docs/articles/baseplatform/booting-qemu-image.rst [hunk 8]
```diff
@@ -41,6 +44,56 @@ following command:
...
+Run the following command on the host machine:
+
+.. code-block:: bash
+
+   sudo ip route add to 192.168.7.1 dev tap0 ; \
+   sudo iptables -A POSTROUTING -t nat -j MASQUERADE -s 192.168.7.1/24; \
```
Comment (mardy) [inline docs/articles/baseplatform/booting-qemu-image.rst]: Same here, I think you can do without ";" and "\\".

**Review by mardy — CHANGES_REQUESTED** [review]

File: docs/articles/baseplatform/booting-qemu-image.rst (hunk 4 above)
Comment (BassemMohsen) [reply docs/articles/baseplatform/booting-qemu-image.rst]: "is" actually here makes sense

File: docs/articles/baseplatform/booting-qemu-image.rst [hunk 9]
```diff
@@ -39,7 +42,56 @@ following command:
       -no-reboot \
       -soundhw ac97 \
       -m 512 \
-      --append "vga=0 uvesafb.mode_option=1280×720-32 root=/dev/hda console=ttyS0 rw mem=512M oprofile.timer=1 " -serial stdio
+      -append "vga=0 uvesafb.mode_option=1280×720-32 root=/dev/hda console=ttyS0 rw mem=512M oprofile.timer=1 " -serial stdio
+
+Static IP address configuration
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+A static IP address facilitates development of automation and testing scripts.
+The next commands assumes target ip address is ``192.168.7.2`` and host machine
```
Comment (mardy) [inline docs/articles/baseplatform/booting-qemu-image.rst]: s/assumes/assume/

File: docs/articles/baseplatform/booting-qemu-image.rst [hunk 10]
```diff
@@ -39,7 +42,56 @@ following command:
...
+commands on the host machine:
+
+.. code-block:: bash
+
+   sudo ip address add 192.168.7.1/24 broadcast 192.168.7.255 dev tap0
+   sudo ip link set dev tap0 up
+   
+
+3. (Optional) Enable the tap device to access internet by running the following
+command on the host machine:
```
Comment (mardy) [inline docs/articles/baseplatform/booting-qemu-image.rst]: s/command/commands/

File: docs/articles/baseplatform/booting-qemu-image.rst [hunk 11]
```diff
@@ -61,14 +113,29 @@ KVM option    Value           Description
 Connecting to the virtual machine using SSH
 -------------------------------------------
 
-The machine can be accessed via SSH for development purposes using the
-following command:
+The machine can be accessed via SSH for development purposes.
+
+SSH access via Port forwarding:
```
Comment (mardy) [inline docs/articles/baseplatform/booting-qemu-image.rst]: You forgot to remove the colon here

File: docs/articles/baseplatform/booting-qemu-image.rst [hunk 12]
```diff
@@ -61,14 +113,29 @@ KVM option    Value           Description
...
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Boot the qemu machine with port forwarding configuration, and then run the
+following command on host machine.
 
 .. code-block:: bash
 
     $ ssh root@127.0.0.1 -p 1234
 
+
+SSH access via static ip:
```
Comment (mardy) [inline docs/articles/baseplatform/booting-qemu-image.rst]: and here

**Review by mardy — CHANGES_REQUESTED** [review]

File: docs/articles/baseplatform/booting-qemu-image.rst (hunk 9 above)
Comment (tobsan) [inline docs/articles/baseplatform/booting-qemu-image.rst]: Can we be more specific when we say "target" and talk about a QEMU target. As a developer, this is confusing, so I assume that a regular user would also be confused by it?

File: docs/articles/baseplatform/booting-qemu-image.rst [hunk 13]
```diff
@@ -61,14 +113,29 @@ KVM option    Value           Description
...
+The machine can be accessed via SSH for development purposes.
+
+SSH access via Port forwarding:
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Boot the qemu machine with port forwarding configuration, and then run the
+following command on host machine.
 
 .. code-block:: bash
 
     $ ssh root@127.0.0.1 -p 1234
```
Comment (tobsan) [inline docs/articles/baseplatform/booting-qemu-image.rst]: Can we add an explanation for this so that users understand that 1234 map to 22?

**Review by tobsan — CHANGES_REQUESTED** [review]

**Review by mardy — APPROVED** [review]

**Review by tobsan — APPROVED** [review]

**Pull request merged**